pyocd/coresight/cortex_m.py

```diff
@@ -524,13 +529,15 @@ def _check_for_fpu(self):
             # Now check whether double-precision is supported.
             mvfr0 = self.read32(CortexM.MVFR0)
             dp_val = (mvfr0 & CortexM.MVFR0_DOUBLE_PRECISION_MASK) >> CortexM.MVFR0_DOUBLE_PRECISION_SHIFT
-            self.has_fpu_double = (dp_val == 2)
+            self.has_fpu_double = (dp_val >= 2)
 
-            if self.core_type == ARM_CortexM7:
-                if self.has_fpu_double:
-                    fpu_type = "FPv5-DP"
-                else:
-                    fpu_type = "FPv5-SP"
+            mvfr2 = self.read32(CortexM.MVFR2)
+            vfp_misc_val = (mvfr2 & CortexM.MVFR2_VFP_MISC_MASK) >> CortexM.MVFR2_VFP_MISC_SHIFT
+
+            if self.has_fpu_double:
+                fpu_type = "FPv5"
```

Comment:
Can we change this to "FPv5-DP"?

Reply:
Could do, and it does seem more explicit, so I'd prefer it, but all the architectural references I can find call it just "FPv5", with the "-SP" as a possible restriction limiter.

The only reference to "FPv5-DP" I can find is in the Cortex-M7 TRM, but it seems to me that may be effectively an error.

(PyOCD actually doesn't need to care in the slightest - double arithmetic support or not has no impact on a debugger, as per #483. That removes `self.has_fpu_double`.)

Reply:
I must have been looking at the CM7 TRM because I do remember seeing "FPv5-DP" in Arm docs somewhere. :)